flink-tests/src/test/java/org/apache/flink/test/state/ChangelogCompatibilityITCase.java

```diff
@@ -169,18 +160,32 @@ private String tryCheckpointAndStop(JobGraph jobGraph) throws Exception {
         }
     }
 
-    private void restoreAndValidate(String location) {
+    private void restoreAndValidate(String location) throws Exception {
         StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.enableChangelogStateBackend(testCase.restoreWithChangelog);
         JobGraph jobGraph = addGraph(env);
         jobGraph.setSavepointRestoreSettings(forPath(location));
 
-        if (tryRun(jobGraph) != testCase.allowRestore) {
```

Comment:
I think `tryRun(jobGraph) != testCase.allowRestore` contains two case of:
`restored && !testCase.allowRestore` and `!restored && testCase.allowRestore`. 
why split it into two.